src/main/java/uk/gov/hmcts/darts/arm/service/impl/AbstractArmBatchProcessResponseFiles.java

```diff
@@ -0,0 +1,836 @@
+                }
+            }
+            if (externalObjectDirectoryEntities.size() == completedExternalObjectDirectoryEntities.size()) {
+                log.info("About to delete ARM input upload file {}", batchUploadFileFilenameProcessor.getBatchMetadataFilename());
+                armDataManagementApi.deleteBlobData(batchUploadFileFilenameProcessor.getBatchMetadataFilenameAndPath());
+            } else {
+                log.warn("Unable to delete ARM batch input upload file {} as referenced data is not complete - total {} vs completed {}",
+                         batchUploadFileFilenameProcessor.getBatchMetadataFilename(),
+                         externalObjectDirectoryEntities.size(), completedExternalObjectDirectoryEntities.size());
+            }
+        } else {
+            // If no EODs are found for the manifest, delete the input upload blob and any linked response files as they are dangling
+            deleteDanglingResponses(batchUploadFileFilenameProcessor);
+        }
+
+    }
+
+    // Delete the response files if they are not linked to any EODs
+    private void deleteDanglingResponses(BatchInputUploadFileFilenameProcessor batchUploadFileFilenameProcessor) {
+        List<String> responseFiles = new ArrayList<>();
+        try {
+            responseFiles.addAll(armDataManagementApi.listResponseBlobs(batchUploadFileFilenameProcessor.getHashcode()));
+        } catch (Exception e) {
+            log.error("Unable to find dangling response files for hashcode {}", batchUploadFileFilenameProcessor.getHashcode(), e);
+        }
+
+        if (CollectionUtils.isNotEmpty(responseFiles)) {
+            List<Boolean> deletedResponseBlobStatuses = deleteResponseBlobs(responseFiles);
+
+            if (!deletedResponseBlobStatuses.contains(false)) {
+                log.info("About to delete ARM input upload file {}", batchUploadFileFilenameProcessor.getBatchMetadataFilename());
+                armDataManagementApi.deleteBlobData(batchUploadFileFilenameProcessor.getBatchMetadataFilenameAndPath());
+            } else {
+                log.warn("Unable to delete ARM batch input upload file {} as referenced data is not all deleted",
+                         batchUploadFileFilenameProcessor.getBatchMetadataFilename());
+            }
+        } else {
+            log.info("About to delete dangling ARM input upload file {}", batchUploadFileFilenameProcessor.getBatchMetadataFilename());
+            armDataManagementApi.deleteBlobData(batchUploadFileFilenameProcessor.getBatchMetadataFilenameAndPath());
+        }
+    }
+
+    private boolean isResponseCompletedAndCleaned(ExternalObjectDirectoryEntity externalObjectDirectory) {
+        return externalObjectDirectory.isResponseCleaned()
+            && isCompletedStatus(externalObjectDirectory.getStatus());
+    }
+
+    private boolean isCompletedStatus(ObjectRecordStatusEntity status) {
+        if (nonNull(status)) {
```

Comment:
can this be 
if(status == null) {
 return false;
}

To avoid nesting

Reply:
changes to existing code that are not necessary for plugging-in DETS functionality cannot be tackled in this PR, as explained in the PR description